_spark/df_api_ex.ipynb

```diff
@@ -0,0 +1,936 @@
+      "| 72397417|2016-05-16|                t|          48.8657|           2.35047|  Entire rental unit|Entire home/apt|           4|      1.0|        1 bath|     1.0| 3.0|[\"Cooking basics\"...|200.0|             2|            30|               12|                4.92|       7510211400994|               f|             1.61|\n",
+      "| 35698627|2015-06-13|                f|          48.8794|           2.32824|  Entire rental unit|Entire home/apt|           1|      1.0|        1 bath|     1.0| 1.0|[\"Room-darkening ...| 36.0|           150|           300|                5|                 4.0|Available with a ...|               f|             0.08|\n",
+      "|  2526137|2012-06-02|              NaN|         48.86334|           2.35045|  Entire rental unit|Entire home/apt|           6|      1.0|        1 bath|     1.0| 3.0|[\"Free dryer \\u20...|214.0|             2|            90|              481|                4.75|       7510201036545|               t|              4.6|\n",
+      "| 21031735|2014-09-07|                f|         48.84015|           2.26161|  Entire rental unit|Entire home/apt|           6|      3.0|       3 baths|     3.0| 7.0|[\"Host greets you...|359.0|             3|            28|               27|                4.48|       7511604873401|               f|             0.38|\n",
+      "+---------+----------+-----------------+-----------------+------------------+--------------------+---------------+------------+---------+--------------+--------+----+--------------------+-----+--------------+--------------+-----------------+--------------------+--------------------+----------------+-----------------+\n",
+      "only showing top 20 rows\n",
+      "\n"
+     ]
+    },
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "                                                                                \r"
+     ]
+    }
+   ],
+   "source": [
+    "clean_df = fixed_price_df.dropna()\n",
+    "clean_df.show()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 21,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "24/11/05 16:12:36 WARN TaskSetManager: Stage 27 contains a task of very large size (41281 KiB). The maximum recommended task size is 1000 KiB.\n"
+     ]
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "+-------+-----------------+\n",
+      "|summary|            price|\n",
+      "+-------+-----------------+\n",
+      "|  count|            45916|\n",
+      "|   mean|212.8508363097831|\n",
+      "| stddev|404.3842515189901|\n",
+      "|    min|              8.0|\n",
+      "|    max|          24200.0|\n",
+      "+-------+-----------------+\n",
+      "\n"
+     ]
+    }
+   ],
+   "source": [
+    "clean_df = clean_df.repartition(100) \n",
```

Comment:
You can change it to `coalesce(100)`, as `repartition` will force full shuffling. I don't think this is the case for 1 node spark cluster, but makes sense in bigger clusters.
https://stackoverflow.com/questions/31610971/spark-repartition-vs-coalesce